Samples/Islands/Lottie/LottieIsland/LottieContentIsland.cpp

```diff
@@ -0,0 +1,275 @@
+            Resize(m_island.ActualSize());
+
+            StartAnimation(0.0, 1.0, true /*loop*/);
+        }
+    }
+
+    winrt::Windows::Foundation::TimeSpan LottieContentIsland::Duration() const
+    {
+        if (m_animatedVisual == nullptr)
+        {
+            return 0ms;
+        }
+
+        return m_animatedVisual.Duration();
+    }
+
+    bool LottieContentIsland::IsAnimationLoaded() const
+    {
+        // Revisit this when we get JSON loading to work.
+        return m_animatedVisual != nullptr;
+    }
+
+    bool LottieContentIsland::IsPlaying() const
+    {
+        return m_progressPropertySet != nullptr;
+    }
+
+    float LottieContentIsland::PlaybackRate() const
+    {
+        return m_playbackRate;
+    }
+
+    void LottieContentIsland::PlaybackRate(float rate)
+    {
+        m_playbackRate = rate;
+        if (m_animationController != nullptr)
+        {
+            m_animationController.PlaybackRate(m_playbackRate);
+        }
+    }
+
+    void LottieContentIsland::Pause()
+    {
+        if (m_animationController != nullptr)
+        {
+            m_animationController.Pause();
+        }
+    }
+
+    winrt::Windows::Foundation::IAsyncAction LottieContentIsland::PlayAsync(float fromProgress, float toProgress, bool looped)
+    {
+        if (m_animationCompletionEvent.get() == nullptr)
+        {
+            m_animationCompletionEvent = winrt::handle(CreateEvent(nullptr, false, false, nullptr));
+        }
+
+        // Stop any existing animation
+        StopAnimation();
+
+        auto batch = m_compositor.CreateScopedBatch(CompositionBatchTypes::Animation);
+
+        StartAnimation(fromProgress, toProgress, looped);
+
+        // Keep track of whether the animation is looped, since we will have to
+        // manually fire the event if Stop() is called in the non-looped case.
+        // We don't hook up the event here in the looped case, because ScopedBatches
+        // complete immediately if their animation is looped.
+        m_looped = looped;
+        if (!looped)
+        {
+            // Hook up an event handler to the Completed event of the batch
+            auto eventToken = batch.Completed([&](auto&&, auto&&)
+                {
+                    // Set the completion event when the batch completes
+                    SetEvent(m_animationCompletionEvent.get());
+                });
+        }
+
+        // Commit the batch
+        batch.End();
+
+        // Wait for the completion event asynchronously
+        co_await winrt::resume_on_signal(m_animationCompletionEvent.get()); // Wait for the event to be signaled
+    }
+
+    void LottieContentIsland::Resume()
+    {
+        if (m_animationController != nullptr)
+        {
+            m_animationController.Resume();
+        }
+    }
+
+    void LottieContentIsland::Stop()
+    {
+        StopAnimation();
+        if (m_looped)
+        {
+            SetEvent(m_animationCompletionEvent.get());
```

Comment:
I think this might have some race conditions - if we call `Stop` first before calling PlayAsync then the event will be set and the async action won't wait at all. Maybe a fix is as simple as a check for IsPlaying?